clients/src/main/java/org/apache/kafka/common/network/Selector.java

```diff
@@ -314,6 +313,9 @@ public void register(String id, SocketChannel socketChannel) throws IOException
         ensureNotRegistered(id);
         registerChannel(id, socketChannel, SelectionKey.OP_READ);
         this.sensors.connectionCreated.record();
+        // Default to empty client information as the ApiVersionsRequest is not
+        // mandatory. In this case, we still want to account for the connection.
+        this.channel(id).channelMetadataRegistry().registerClientInformation(ClientInformation.EMPTY);
```

Comment:
I think it would be good to check that the registry doesn't already have something registered before we do this (since otherwise we'll be overwriting whatever that is).

Reply:
Good point. It should not happen but it is better to be on the safe side.